valhalla/jawn/src/lib/shared/delayedOperationService.ts

```diff
@@ -0,0 +1,73 @@
+export class DelayedOperationService {
+  private static instance: DelayedOperationService;
+  private delayedOperations: Map<NodeJS.Timeout, () => Promise<any>> =
+    new Map();
+  private static readonly SHUTDOWN_TIMEOUT = 60000; // 60 seconds timeout
```

Comment:
Do we need more time than this?

Reply:
@chitalian We can do 3 minutes actually, thats more than enough

Reply:
We will keep shutdown timeout 30 seconds because it works only for local env, AWS just shutting down whole container :(